cmd/thanos/receive.go

```diff
@@ -436,23 +439,24 @@ func setupHashring(g *run.Group,
 	webHandler *receive.Handler,
 	statusProber prober.Probe,
 	enableIngestion bool,
-) error {
+) ([]receive.HashringConfig, error) {
 	// Note: the hashring configuration watcher
 	// is the sender and thus closes the chan.
 	// In the single-node case, which has no configuration
 	// watcher, we close the chan ourselves.
 	updates := make(chan receive.Hashring, 1)
 	algorithm := receive.HashringAlgorithm(conf.hashringsAlgorithm)
+	var hashringConfigs []receive.HashringConfig
 
 	// The Hashrings config file path is given initializing config watcher.
 	if conf.hashringsFilePath != "" {
 		cw, err := receive.NewConfigWatcher(log.With(logger, "component", "config-watcher"), reg, conf.hashringsFilePath, *conf.refreshInterval)
 		if err != nil {
-			return errors.Wrap(err, "failed to initialize config watcher")
+			return nil, errors.Wrap(err, "failed to initialize config watcher")
 		}
 
 		// Check the hashring configuration on before running the watcher.
-		if err := cw.ValidateConfig(); err != nil {
+		if hcs, err := cw.ValidateConfig(); err != nil {
```

Comment:
Same `hcs` not defined.
```suggestion
		hcs, err := cw.ValidateConfig()
		if err != nil {
```